core/src/test/java/com/radixdlt/rev2/LedgerProofsGcTest.java

```diff
@@ -114,36 +129,42 @@ private DeterministicTest createTest(long mostRecentFullResolutionEpochCount, lo
                 true,
                 SafetyRecoveryConfig.MOCKED,
                 ConsensusConfig.of(1000),
-                LedgerConfig.stateComputerNoSync(
+                LedgerConfig.stateComputerWithSyncRelay(
                     new StateComputerConfig.REv2StateComputerConfig(
                         Network.INTEGRATIONTESTNET.getId(),
                         GenesisBuilder.createTestGenesisWithNumValidators(
                             1,
                             Decimal.ONE,
                             GenesisConsensusManagerConfig.Builder.testWithRoundsPerEpoch(
-                                epochLength)),
+                                roundsPerEpoch)),
                         new DatabaseFlags(false, false),
                         REV2ProposerConfig.transactionGenerator(
-                            new MbTransactionGenerator(NetworkDefinition.INT_TEST_NET), 3),
+                            new SizedTransactionGenerator(NetworkDefinition.INT_TEST_NET, txnSize),
+                            1),
                         false,
                         StateHashTreeGcConfig.forTesting(),
                         new LedgerProofsGcConfig(
                             UInt32.fromNonNegativeInt(GC_INTERVAL_SEC),
                             UInt64.fromNonNegativeLong(mostRecentFullResolutionEpochCount)),
-                        false))));
+                        new LedgerSyncLimitsConfig(
+                            UInt32.fromNonNegativeInt(maxTxnCountUnderProof),
+                            UInt32.fromNonNegativeInt(maxTxnPayloadSizeUnderProof)),
+                        false),
+                    SyncRelayConfig.of(100, 2, 200L))));
   }
 
   @Test
-  public void node_prunes_non_critical_proofs_from_configured_old_enough_epochs() {
-    // Arrange: configure ledger proof gc to keep 2 most recent completed epochs not pruned
-    var mostRecentFullResolutionEpochCount = 2;
-    /// ... 6 rounds * 3 transactions * 1 MB = 18MB; Limit is 12 MB - so we require 2 proofs
-    var epochLength = 6;
-    try (var test = createTest(mostRecentFullResolutionEpochCount, epochLength)) {
-      test.startAllNodes();
+  public void node_retains_enough_proofs_to_cover_max_transaction_size_in_old_enough_epochs() {
+    /// 6 rounds * 100 KB = 600KB
+    var roundsPerEpoch = 6;
+    var txnSize = 100 * 1024;
+    /// Limit is 400 KB - so we require 2 proofs to cover 600 KB
+    var maxTxnPayloadSizeUnderProof = 400 * 1024; // cannot be much lower due to genesis' size
```

Comment:
Thanks for the comment!

This is a really good point, and raises a possible issue. Big gaps in genesis are irrelevant for ledger sync, because genesis never needs to be synced... BUT it is a problem for the pruner unless we special case genesis. In particular, on mainnet, we have no proofs across all of genesis.

I think we may have to special case genesis in the pruner (basically ignore it / don't try to clear up proofs). This might also apply to protocol update transactions, but we can ignore that until discussions shortly.

Reply:
Huh, that's all true observation, but addressing it is larger than expected.
Let's track it as [another work item](https://docs.google.com/spreadsheets/d/1JPiisoLFOhrDBT1-qDj2uvVwIsojKQd8kHzf1CIhaFw/edit#gid=0&range=B19), and another PR.